CMakeLists.txt

```diff
@@ -1,174 +1,176 @@
+    target_precompile_headers(${TARGET-NAME} REUSE_FROM mbed-os)
+endfunction()
+
+# link against mbed and setup linker script
+function(target_set_firmware_properties TARGET-NAME)
+    target_set_mbed_dependency(${TARGET-NAME})
+
+    set_target_properties(${TARGET-NAME} PROPERTIES LINK_DEPENDS ${LINKER_SCRIPT_OUTPUT_PATH})
+    add_dependencies(${TARGET-NAME} linker-script)
+
+    target_bin_from_elf(${TARGET-NAME})
+endfunction()
+
+# Global Settings
+set(COMPILE_FLAGS
+        -MMD
+        -MP
+
+        -O3
+        -g
+
+        -Wall
+        -Wextra
+        -Wpedantic
+        -Wvla
+        -Werror
+
+        -Wno-unused-parameter
+        -Wno-missing-field-initializers
+
+        -ffunction-sections
+        -fdata-sections
+        -fomit-frame-pointer
+        -funsigned-char
+
+        -fno-exceptions
+        -fno-delete-null-pointer-checks
+
+        -fstack-protector-strong
+
+        $<$<COMPILE_LANGUAGE:ASM>:-xassembler-with-cpp>
+        $<$<COMPILE_LANGUAGE:CXX>:-fno-rtti>
+        )
+add_compile_options(${PROCESSOR_FLAGS} ${COMPILE_FLAGS})
+
+# Include mbed-os
+include(mbed-os.cmake)
+
+# Include library CMakeLists
+add_subdirectory(lib/can)
+add_subdirectory(lib/gamepad)
+add_subdirectory(lib/controllers)
+add_subdirectory(lib/lookup-table)
+add_subdirectory(lib/neopixel)
+add_subdirectory(lib/pid)
+add_subdirectory(lib/sensors)
+add_subdirectory(lib/servo)
+
+if (NOT DEFINED APP)
+    message(FATAL_ERROR "APP variable not set in CACHE. Please invoke CMake with \"-DAPP=<app-name>\"")
+elseif (NOT EXISTS "${CMAKE_SOURCE_DIR}/apps/${APP}")
+    message(FATAL_ERROR "${APP} app does not exist in the apps folder")
+endif ()
+if (NOT DEFINED TARGET)
+    message(FATAL_ERROR "TARGET variable not set in CACHE. Please invoke CMake with \"-DTARGET=<target-name>\"")
+elseif (NOT EXISTS "${CMAKE_SOURCE_DIR}/targets/${TARGET}")
+    message(FATAL_ERROR "${TARGET} target does not exist in the targets folder")
+endif ()
+message(STATUS "Configuring and generating build for ${APP} app on ${TARGET} target")
+
+# Add CMakeList for target-specific files
+add_subdirectory(targets/${TARGET})
+
+# Add HW Bridge
+add_subdirectory(uwrt-mars-rover-hw-bridge)
+
+# Include Rover App CMakeLists
+add_subdirectory(apps/arm)
+add_subdirectory(apps/gamepad)
+add_subdirectory(apps/gimbtonomy)
+add_subdirectory(apps/pdb)
+add_subdirectory(apps/science)
+
+# Include Test App CMakeLists
+add_subdirectory(apps/test-actuator-controller)
+add_subdirectory(apps/test-aeat-8800)
+add_subdirectory(apps/test-blinky)
+add_subdirectory(apps/test-blockingneo)
+add_subdirectory(apps/test-can)
+add_subdirectory(apps/test-enc-abs-pwm)
+add_subdirectory(apps/test-lookup-table)
+add_subdirectory(apps/test-moisture)
+add_subdirectory(apps/test-pid)
+add_subdirectory(apps/test-pwm)
+add_subdirectory(apps/test-pwmin)
+add_subdirectory( apps/tutorial-servo-pot-control )
+add_subdirectory( apps/tutorial-servo-can-control )

```

Comment:
It's weird that this is showing up as a whole file change... if it's just a fine lines that you modified it should only show those changes

Reply:
Yeah I don't know why that's the case